src/xmppjs/XJSGateway.ts

```diff
@@ -419,10 +443,43 @@ export class XmppJsGateway implements IGateway {
 
         // 4. Room history
         log.debug("Emitting history");
-        const history: Element[] = this.roomHistory.get(room.roomId) || [];
+        const historyLimits: IHistoryLimits = {};
+        const historyRequest = stanza.getChild("x", "http://jabber.org/protocol/muc")?.getChild("history");
+        if (historyRequest !== undefined) {
+            const getIntValue = (str) => {
+                if (!str.match(/^\d+$/)) {
```

Comment:
No need to `match` if you only care for a boolean return value.
```suggestion
                if (!/^\d+$/.test(str)) {
```